### ISSUE TITLE
False positive `let_var_whitespace` violation when let/var follows multiline attribute

### ISSUE THREAD

**Issue by tinder-garricnahapetian** [realm/SwiftLint#5483]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

SwiftLint triggers a false positive `let_var_whitespace` for a let/var that follows a multiline attribute.

```bash
SwiftLint on  main [$] via 💎 ruby-3.1.0 
➜ swiftlint --version                                        
0.54.0

SwiftLint on  main [$] via 💎 ruby-3.1.0 
➜ swiftlint lint --quiet --progress --reporter emoji         
warning: Invalid configuration for 'unneeded_override'. Falling back to default.
warning: 'one_declaration_per_file' is not a valid rule identifier
warning: 'final_test_case' is not a valid rule identifier
warning: 'shorthand_argument' is not a valid rule identifier
Valid rule identifiers:
...rules..

/Users/garricnahapetian/Developer/SwiftLint/Source/SwiftLintBuiltInRules/Rules/RuleConfigurations/DiscouragedDirectInitConfiguration.swift
⚠️ Line 14: Variable declarations should be separated from other statements by a blank line (let_var_whitespace)
/Users/garricnahapetian/Developer/SwiftLint/Source/SwiftLintBuiltInRules/Rules/RuleConfigurations/IndentationWidthConfiguration.swift
⚠️ Line 13: Variable declarations should be separated from other statements by a blank line (let_var_whitespace)
⚠️ Line 9: Variable declarations should be separated from other statements by a blank line (let_var_whitespace)
/Users/garricnahapetian/Developer/SwiftLint/Source/SwiftLintBuiltInRules/Rules/RuleConfigurations/TestCaseAccessibilityConfiguration.swift
⚠️ Line 15: Variable declarations should be separated from other statements by a blank line (let_var_whitespace)
⚠️ Line 11: Variable declarations should be separated from other statements by a blank line (let_var_whitespace)
/Users/garricnahapetian/Developer/SwiftLint/Source/SwiftLintBuiltInRules/Rules/RuleConfigurations/UnitTestConfiguration.swift
⚠️ Line 17: Variable declarations should be separated from other statements by a blank line (let_var_whitespace)
⚠️ Line 13: Variable declarations should be separated from other statements by a blank line (let_var_whitespace)
/Users/garricnahapetian/Developer/SwiftLint/Source/SwiftLintBuiltInRules/Rules/RuleConfigurations/UnusedDeclarationConfiguration.swift
⚠️ Line 15: Variable declarations should be separated from other statements by a blank line (let_var_whitespace)
⚠️ Line 11: Variable declarations should be separated from other statements by a blank line (let_var_whitespace)
/Users/garricnahapetian/Developer/SwiftLint/Source/swiftlint/Helpers/LintOrAnalyzeArguments.swift
⚠️ Line 28: Variable declarations should be separated from other statements by a blank line (let_var_whitespace)
/Users/garricnahapetian/Developer/SwiftLint/Tests/SwiftLintFrameworkTests/RuleConfigurationDescriptionTests.swift
⚠️ Line 31: Variable declarations should be separated from other statements by a blank line (let_var_whitespace)
⚠️ Line 27: Variable declarations should be separated from other statements by a blank line (let_var_whitespace)
```

I believe this [commit](https://github.com/realm/SwiftLint/commit/8cb2cf9f8524a6a8d6c9151b885605a90e4bcde3) introduced the issue. In my testing, the issue is NOT reproducible prior to that commit.

### Example

The following example is taken from the observed violation [here](https://github.com/realm/SwiftLint/blob/1aa8289085c1ba35c192d297254da4e8b6bbebc6/Source/SwiftLintBuiltInRules/Rules/RuleConfigurations/DiscouragedDirectInitConfiguration.swift#L14).
 
False positive violating example:

```swift
@ConfigurationElement(
    key: "types",
    postprocessor: { ... }
)
private(set) var discouragedInits: Set = [ // let_var_whitespace violation
    "Bundle",
    "NSError",
    "UIDevice"
]
```

The violation can be avoided by reformatting as follows::

```swift
@ConfigurationElement(key: "types", postprocessor: { ... })
private(set) var discouragedInits: Set = [
    "Bundle",
    "NSError",
    "UIDevice"
]
```

**Comment by SimplyDanny** [comment]
Can you provide an example that triggers the rule while it shouldn't?

**Comment by tinder-garricnahapetian** [comment]
> Can you provide an example that triggers the rule while it shouldn't?

Thank you @SimplyDanny. I've updated the description with an example section. Let me know if you need anything else.

**Comment by SimplyDanny** [comment]
> > Can you provide an example that triggers the rule while it shouldn't?
> 
> 
> 
> Thank you @SimplyDanny. I've updated the description with an example section. Let me know if you need anything else.

Hm ... there is something odd with the closing braces/parentheses in the examples. Please check them again, @tinder-garricnahapetian.

**Comment by tinder-garricnahapetian** [comment]
> > > Can you provide an example that triggers the rule while it shouldn't?
> > 
> > 
> > Thank you @SimplyDanny. I've updated the description with an example section. Let me know if you need anything else.
> 
> Hm ... there is something odd with the closing braces/parentheses in the examples. Please check them again, @tinder-garricnahapetian.

Updated. Had an extra ` })`. The violation is occurring in the SwiftLint repo so it might be easier to use the actual code as the example which is link in the description and which I will link again [here](https://github.com/realm/SwiftLint/blob/1aa8289085c1ba35c192d297254da4e8b6bbebc6/Source/SwiftLintBuiltInRules/Rules/RuleConfigurations/DiscouragedDirectInitConfiguration.swift#L14).

**Comment by SimplyDanny** [comment]
Now I get it. Didn't notice you're referring to the SwiftLint repository itself.

Linting the SwiftLint repository with SwiftLint built from the same sources should never report any violations. Otherwise, the built would have failed. So if you run `swift run swiftlint`, there shall never be any issues. And this is still the case.

I assume, you are linting SwiftLint with an older version (possibly the Homebrew version?). It reports these warnings. The commit you are pointing to actually fixes the issues with multiline attributes.

**Comment by tinder-garricnahapetian** [comment]
> Now I get it. Didn't notice you're referring to the SwiftLint repository itself.
> 
> Linting the SwiftLint repository with SwiftLint built from the same sources should never report any violations. Otherwise, the built would have failed. So if you run `swift run swiftlint`, there shall never be any issues. And this is still the case.
> 
> I assume, you are linting SwiftLint with an older version (possibly the Homebrew version?). It reports these warnings. The commit you are pointing to actually fixes the issues with multiline attributes.

I understand. My bad. Sorry for not making that more clear from the outset. This must mean that the preferred way to lint the SwiftLint repo using SwiftLint is via `swift run swiftlint` as opposed to using the `SwiftLintPlugin` or anything using the `SwiftLintBinary` which will likely always be out of date with `main`.

I would like to be able to benefit from the `SwiftLintPlugin` in Xcode while working in the SwiftLint repo but this brings to light a potential issue. I will think about this for a bit and create a new issue if needed. I think this issue can be closed for now. Thank you!

**Comment by SimplyDanny** [comment]
> This must mean that the preferred way to lint the SwiftLint repo using SwiftLint is via `swift run swiftlint` as opposed to using the `SwiftLintPlugin` or anything using the `SwiftLintBinary` which will likely always be out of date with `main`.

Yes, plugins can only run pre-built binaries. It's not (yet) allowed to compile something before. So you are right, that the plugin always uses an outdated (in the context of the SwiftLint repository) SwiftLint version. `swift run` works, but also just running `IntegrationTests` or one of its test methods.
